pkg/reconciler/kubernetes/internal/kubeclient.go

```diff
@@ -132,24 +137,32 @@ func (kube *KubeClient) ApplyWithNamespaceOverride(u *unstructured.Unstructured,
 		if err != nil {
 			return metadata, err
 		}
-
-		metadata.Name = u.GetName()
-		metadata.Namespace = u.GetNamespace()
-		metadata.Kind = u.GroupVersionKind().Kind
 		return metadata, nil
 	}
 
-	replace := newReplace(helper)
-	replacedObject, err := replace(u, u.GetNamespace(), u.GetName())
-	if err != nil {
+	updateStrategyResolver := newDefaultUpdateStrategyResolver(helper)
+
+	strategy, err := updateStrategyResolver.Resolve(u)
+	if err != nil || strategy == SkipUpdateStrategy {
```

Comment:
Why do we return an error when a SkipStragey was replied?

Reply:
I can separate those

Reply:
done